src/main/java/com/cjcrafter/armormechanics/listeners/DurabilityListener.kt

```diff
@@ -0,0 +1,75 @@
+package com.cjcrafter.armormechanics.listeners
+
+import com.cjcrafter.armormechanics.durability.DurabilityManager.changeDurability
+import com.cjcrafter.armormechanics.durability.DurabilityManager.getDurability
+import me.deecaad.core.compatibility.CompatibilityAPI
+import org.bukkit.Material
+import org.bukkit.event.EventHandler
+import org.bukkit.event.EventPriority
+import org.bukkit.event.Listener
+import org.bukkit.event.inventory.PrepareAnvilEvent
+import org.bukkit.event.inventory.PrepareItemCraftEvent
+import org.bukkit.event.player.PlayerItemDamageEvent
+import org.bukkit.event.player.PlayerItemMendEvent
+import org.bukkit.inventory.ItemStack
+
+class DurabilityListener : Listener {
+    @EventHandler(priority = EventPriority.HIGH, ignoreCancelled = true)
+    fun onItemDamaged(event: PlayerItemDamageEvent) {
+        changeDurability(event.item, -event.damage)
+    }
+
+    @EventHandler(priority = EventPriority.HIGH, ignoreCancelled = true)
+    fun onItemMend(event: PlayerItemMendEvent) {
+        changeDurability(event.item, event.repairAmount)
+    }
+
+    @EventHandler
+    fun onRepairAnvil(event: PrepareAnvilEvent) {
+        var totalDurability = 0
+        val contents = event.inventory.contents.filter{ it != null && it.type != Material.AIR }
+        if (contents.size != 2) {
+            event.result = null
+            return
+        }
+        val resultItem: ItemStack = event.result ?: return
+        val titles = ArrayList<String>()
+        contents.forEach { item ->
+            if (item != null && item.type.maxDurability > 0) {
+                val title =
+                    CompatibilityAPI.getNBTCompatibility().getString(item, "ArmorMechanics", "armor-title")
+                titles.add(title)
+                totalDurability += getDurability(item)
+            }
+        }
+        if (titles.distinct().size != 1) {
```

Comment:
I see 3 or 4 data structures being created and I can't tell what this is supposed to be doing. I also side some side effects for non armor mechanics items...

e.g.
```
        if (contents.size != 2) {
            event.result = null
            return
        }
```
Why set the result to null? Why not just return? Seems like this might interfere with vanilla behavior or other plugins when we don't even know if there was an ArmorMechanics item involved.

e.g.
```kotlin
if (titles.distinct().size != 1)
```
This line and the loop above it uses 2 data structures to accomplish what a simple kotlin optional could handle.

Reply:
Understood the side effect part, will fix that. 

This listener basically does:
For example we have an ArmorMechanics item "X" 
When player tries to repair their X using another X in anvil plugin should return item with correct durability.

The reason I check distinct title size is for being sure they are not trying to fix X using Y (another ArmorMechanics item). 

To get rid of data types I can probably just use one loop that is like: 
for item in event.inventory.items
  Continue if item is Air or Null
  Add title of the item to titles Set (will check set size after loop is done) 
  Total durability += Durability of the item 
Do you think this is what should be done?